crates/vm/levm/src/precompiles.rs

```diff
@@ -95,12 +98,70 @@ pub fn execute_precompile(current_call_frame: &mut CallFrame) -> Result<Bytes, V
     Ok(result)
 }
 
-fn ecrecover(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+fn fill_with_zeros(slice: &[u8]) -> [u8; 128] {
+    let mut result = [0; 128];
+
+    let n = slice.len().min(128);
+
+    let trimmed_slice = slice.get(..n).unwrap_or_default();
+    result
+        .get_mut(..n)
+        .unwrap_or_default()
+        .copy_from_slice(trimmed_slice);
```

Comment:
If there is no other easy way of doing this without using `copy_from_slice` it is ok to keep it like that for now though.

Reply:
In this case it should never panic, since both unwrap_or_default ensure there will be a 128 sized slice on each side. Anyway I'm looking for an alternative

Reply:
Replaced in [Avoid using copy_from_slice since can panic](https://github.com/lambdaclass/ethrex/pull/1511/commits/c3e095a7596b1027dabdcaf05d3ab4b4e83e8f08)